bin/run.js

```diff
@@ -12,6 +12,23 @@ const IGNORED_GLOBS = [
 let QUnit;
 
 function run( args, options ) {
+	let running = true;
+
+	// Default to non-zero exit code to avoid false positives
+	process.exitCode = 1;
+
+	process.on( "exit", function() {
+		if ( running ) {
+			console.error( "Error: Process exited before tests finished running" );
+
+			if ( QUnit.config.current.semaphore ) {
+				console.error( "Last test to run has an async hold. " +
```

Comment:
Could we include the test name or other identifiable information? In some circumstances with TAP you aren't aware of the test that is running until it completes, making tracking down this test a bit harder....